ui-spacetimechart/src/components/OccupancyBlocLayer.tsx

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react';
+
+import { useDraw } from '../hooks/useCanvas';
+import { type DrawingFunction } from '../lib/types';
+
+export type OccupancyBloc = {
+  timeStart: number;
+  timeEnd: number;
+  spaceStart: number;
+  spaceEnd: number;
+};
+
+export type OccupancyBlocLayerProps = {
+  occupancyBlocs: OccupancyBloc[];
+};
+
+const SIGNALS = ['#CAEDDB', '#FFD4D8', '#FFEABF'];
+
+export const OccupancyBlocLayer = ({ occupancyBlocs }: OccupancyBlocLayerProps) => {
+  const drawOccupancyBlocLayer = useCallback<DrawingFunction>(
+    (ctx, { getTimePixel, getSpacePixel }) => {
+      for (let i = 0; i < occupancyBlocs.length; i++) {
+        const occupancyBloc = occupancyBlocs[i];
+        const x = getTimePixel(occupancyBloc.timeStart);
+        const y = getSpacePixel(occupancyBloc.spaceStart);
+        const width = getTimePixel(occupancyBloc.timeEnd) - x;
+        const height = getSpacePixel(occupancyBloc.spaceEnd) - y;
+
+        ctx.fillStyle = SIGNALS[i % SIGNALS.length];
```

Comment:
Is there some kind of a guarantee that the blocks are ordered by signal state? If so, it doesn't seem straightforward so we should document it. But would it make sense to add a `state` property to `OccupancyBloc` instead?

Reply:
it makes sense for those three signals specifically, red -> yellow -> green -> red ... in a normal situation, but we can also add some other signals like blinking-yellow that doesn't follow this logic

Reply:
If another train crosses the projected path and we want to show that train's occupancy blocks, we might only have one yellow/red block, I think?

Reply:
fixed !